pkg/model/azuremodel/network.go

```diff
@@ -41,14 +45,134 @@ func (b *NetworkModelBuilder) Build(c *fi.CloudupModelBuilderContext) error {
 	}
 	c.AddTask(networkTask)
 
+	nsgTask := &azuretasks.NetworkSecurityGroup{
+		Name:          fi.PtrTo(b.NameForVirtualNetwork()),
+		Lifecycle:     b.Lifecycle,
+		ResourceGroup: b.LinkToResourceGroup(),
+		Tags:          map[string]*string{},
+	}
+	var sshAccessIPv4, sshAccessIPv6 []string
+	for _, cidr := range b.Cluster.Spec.SSHAccess {
+		switch net.IPFamilyOfCIDRString(cidr) {
```

Comment:
Oooh - I didn't know about this function, I think we have a lot of places that check if an IP is an IPv4 or IPv6 that could probably benefit from this!

Reply:
Very true. Also we have some of our own util for this that we could replace.